hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/autoassign/AutoAssignChecker.java

```diff
@@ -86,27 +58,16 @@
     public AutoAssignChecker(final TargetFilterQueryManagement targetFilterQueryManagement,
             final TargetManagement targetManagement, final DeploymentManagement deploymentManagement,
             final PlatformTransactionManager transactionManager, final TenantAware tenantAware) {
-        this.targetFilterQueryManagement = targetFilterQueryManagement;
+        super(targetFilterQueryManagement, deploymentManagement, transactionManager, tenantAware);
         this.targetManagement = targetManagement;
-        this.deploymentManagement = deploymentManagement;
-        this.transactionManager = transactionManager;
-        this.tenantAware = tenantAware;
     }
 
     @Override
     @Transactional(propagation = Propagation.REQUIRES_NEW)
     public void check() {
         LOGGER.debug("Auto assigned check call");
 
-        final PageRequest pageRequest = PageRequest.of(0, PAGE_SIZE);
-
-        final Page<TargetFilterQuery> filterQueries = targetFilterQueryManagement.findWithAutoAssignDS(pageRequest);
-
-        // make sure the filter queries are executed in the order of weights
-        for (final TargetFilterQuery filterQuery : filterQueries) {
-            runInUserContext(filterQuery, () -> checkByTargetFilterQueryAndAssignDS(filterQuery));
-        }
-
+        forEachFilterWithAutoAssignDS(this::checkByTargetFilterQueryAndAssignDS);
```

Comment:
seems like you've forgotten `runInUserContext`, right? either add it here, or directly to `forEachFilterWithAutoAssignDS`